lib/rust/ensogl/component/toggle-button/src/lib.rs

```diff
@@ -48,6 +48,33 @@ pub trait ColorableShape: ShapeWithDefaultableData {
     fn set_color(&self, color: color::Rgba);
 }
 
+/// A shape that can be parametrized by any cached icon. Same as
+/// [`display::shape::compound::from_cache::recolorized`], but has an overlay for capturing mouse
+/// events and implements [`ColorableShape`].
+pub mod any_cached {
```

Comment:
Instead of creating a whole new shape, it would be better to implemenet `ColorableShape` for `recolorized`. Same goes for mouse capture overlay. Alternatively, a dedicated rectangle shape could be used for mouse event capture.